packages/react-server/src/ReactFizzServer.js

```diff
@@ -411,11 +417,19 @@ function popComponentStackInDEV(task: Task): void {
   }
 }
 
-function logRecoverableError(request: Request, error: mixed): void {
+function logRecoverableError(request: Request, error: any): ?string {
   // If this callback errors, we intentionally let that error bubble up to become a fatal error
   // so that someone fixes the error reporting instead of hiding it.
   const onError = request.onError;
-  onError(error);
+  let errorMsg = onError(error);
+  if (__DEV__) {
+    const msg = error && error.message ? error.message : error;
+    errorMsg = JSON.stringify({
+      error: msg,
+      componentStack: getCurrentStackInDEV(),
```

Comment:
This stack will no longer be correct by the time we get here because we've already popped the component stacks by the time we get here. So this might only tell you the nearest suspending component or nearest suspense boundary.

To do this accurately, we'd need to add a try/catch around deeper points in the tree and catch it + stash it at those places.

Reply:
Yeah I was thinking that the nearest suspend boundary might be fine but yeah I guess we should stash it and then have it be consumed here